src/pages/permissions/panels/EditGroupPermissionsPanel.tsx

```diff
@@ -0,0 +1,481 @@
+  generateIdentityNamesLookup,
+  generateImageNamesLookup,
+  generatePermissionSort,
+  getPermissionId,
+} from "util/permissions";
+import { useToastNotification } from "context/toastNotificationProvider";
+import NotificationRow from "components/NotificationRow";
+import ScrollableContainer from "components/ScrollableContainer";
+import { fetchImageList } from "api/images";
+import { fetchIdentities } from "api/auth-identities";
+import useEditHistory from "util/useEditHistory";
+import classnames from "classnames";
+import ModifiedStatusAction from "../actions/ModifiedStatusAction";
+import { useSettings } from "context/useSettings";
+import { getIdentitiesForGroup } from "util/permissionIdentities";
+import LoggedInUserNotification from "./LoggedInUserNotification";
+
+type LxdPermissionWithID = LxdPermission & { id: string };
+
+type PermissionsEditHistory = {
+  permissionsAdded: LxdPermissionWithID[];
+  permissionsRemoved: Set<string>;
+};
+
+interface Props {
+  group: LxdGroup;
+}
+
+const EditGroupPermissionsPanel: FC<Props> = ({ group }) => {
+  const { data: settings } = useSettings();
+  const panelParams = usePanelParams();
+  const notify = useNotify();
+  const toastNotify = useToastNotification();
+  const queryClient = useQueryClient();
+  const [submitting, setSubmitting] = useState(false);
+  const [resourceType, setResourceType] = useState("");
+  const [resource, setResource] = useState("");
+  const [entitlement, setEntitlement] = useState("");
+  const [search, setSearch] = useState("");
+  const [confirming, setConfirming] = useState(false);
+  const loggedInUser = settings?.auth_user_name;
+  const { allIdentities } = getIdentitiesForGroup(group);
+  const groupAllocatedToLoggedInUser = allIdentities.some(
+    (identity) => identity === loggedInUser,
+  );
+
+  const {
+    desiredState,
+    save: saveToPermissionHistory,
+    undo: undoPermissionChange,
+  } = useEditHistory<PermissionsEditHistory>({
+    initialState: {
+      permissionsAdded: [],
+      permissionsRemoved: new Set(),
+    },
+  });
+
+  const { data: images = [], isLoading: isImagesLoading } = useQuery({
+    queryKey: [queryKeys.images],
+    queryFn: () => fetchImageList(),
+  });
+
+  const { data: identities = [], isLoading: isIdentitiesLoading } = useQuery({
+    queryKey: [queryKeys.identities],
+    queryFn: fetchIdentities,
+  });
+
+  const resetSelectors = () => {
+    setResourceType("");
+    setResource("");
+    setEntitlement("");
+  };
+
+  const getCurrentPermissions = () => {
+    const existingPermissions = group.permissions ?? [];
+    const currentPermissions: LxdPermissionWithID[] = [];
+    for (const permission of existingPermissions) {
+      currentPermissions.push({
+        ...permission,
+        id: getPermissionId(permission),
+      });
+    }
+
+    for (const permission of desiredState.permissionsAdded) {
+      currentPermissions.push({
+        ...permission,
+        id: getPermissionId(permission),
```

Comment:
```suggestion
```
We don't really need this, do we?

Reply:
nice catch, yes in this case we don't need the id since it's already added.